lib/3scale/backend/worker_async.rb

```diff
@@ -32,13 +31,11 @@ def initialize(options = {})
       def work
         return Sync { process_one } if one_off?
 
-        Sync { register_worker }
-
-        fetch_jobs_thread = start_thread_to_fetch_jobs
-
-        Sync { process_all }
-
-        fetch_jobs_thread.join
+        Sync do
+          register_worker
+          start_to_fetch_jobs
+          process_all
+        end
```

Comment:
We don't care too much about order of these. Also fetcher now should use the same reactor as the job processing.

Reply:
I agree we don't care about the order of `start_to_fetch_jobs` and `process_all`. But I think `register_worker` should be finished before entering this block. Even when we don't know exactly what's for.

Reply:
done